aiinfra-cluster/modules/aiinfra-compute/main.tf

```diff
@@ -78,8 +79,20 @@ locals {
     access_config      = var.disable_public_ips ? [] : [local.empty_access_config]
     ipv6_access_config = []
     alias_ip_range     = []
-  }
-  network_interfaces = coalescelist(var.network_interfaces, [local.default_network_interface])
+  }]
+
+  network_interfaces = coalescelist(var.network_interfaces, local.default_network_interface)
+
+  default_node_pool         = [{
+    name                    = "system-nodes"
+    nodes_initial           = var.instance_count
+    nodes_min               = var.instance_count
+    nodes_max               = var.instance_count
+    machine_type            = var.machine_type
+    guest_accelerator_count = var.guest_accelerator.count
+    guest_accelerator_type  = var.guest_accelerator.type
+  }]
+  gke_node_pools            = coalescelist(var.node_pools, local.default_node_pool)
```

Comment:
Why join here?
I thought we either create default node pool, of use var.node_pools, but never both.

Reply:
The terraform `coalescelist` function behavior is different. It returns the first non-empty list. So here, we are taking default_node_pool only if the var.node_pools is empty.
https://developer.hashicorp.com/terraform/language/functions/coalescelist

Reply:
Thanks